tartufo/scanner.py

```diff
@@ -90,17 +92,26 @@ def file_path(self) -> Optional[str]:
             return self.diff.b_path
         return self.diff.a_path
 
+    @property
+    def signature(self) -> Optional[str]:
+        if self.file_path:
+            return generate_signature(self.matched_string, self.file_path)
+        return None
```

Comment:
Yeah, that kept going through my head as I was working on this code. I think the path toward 2.0 is going to involve a lot of cleanup of issues like this. There really should always be a file path. I just want to be able to trace through the code and confidently say that before I write methods like this without that extra check, just in case.